env/git.go

```diff
@@ -112,26 +112,21 @@ func GitSHA(path string) (string, error) {
 }
 
 var GitBlob = func(path string, commit *object.Commit) (string, error) {
-	if commit == nil {
-		blob, err := fallbackBlob(path)
+	if commit != nil {
+		if file, err := commit.File(path); err == nil {
```

Comment:
This `err` never gets checked. Isn't that the exact problem the PR description is saying you're addressing?

(Knowing the `commit` but not getting the `File` because it's gitignored.)